### DISCUSSION TITLE
Blazor Hot reload: better fallback mechanism, avoid closing the browser and retain the url

### DISCUSSION THREAD

**Discussion by mip1983** [dotnet/aspnetcore#57961] [Ideas]
I've a suggestion for hot-reload with Blazor to give a less interrupted experience and keep you more in the flow, giving a reload experience similar to angular:

At the moment if you change something and it cannot hot-reload it in, you get a pop-up and need to stop and restart debugging which closes the browser.

Would it be possible in this situation instead to put up a similar prompt to let the user know, but on reload:

1. Keep the browser window open
2. Suspend the app with an overlay or holding page (like when the signalR connection drops)
3. Do the full compile
4. Once compiled/restarted, reload the browser window retaining the original path.

That gives a less interrupted experience. You'll lose the state of your page (unless you've coded something to retain it in storage), but will at least be back on the page you where working on with the app re-loaded.

Bonus:

In a .NET Aspire context it could work the same as above, but keep the Aspire host running as well, and recompile/restart the container apps in the host.

**Comment by timdinh** [comment]
Hot reload with Blazor is terrible compared with other framework like vue.js. I think the team need to focus on this issue as high priority otherwise developers will drop this framework fast.

**Reply by mip1983** [reply]
Looks like they are: https://www.youtube.com/live/fye6bN1-TT4?t=2432s
